src/Composer.jsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import { push, set, ref as databaseRef } from "firebase/database";
+import {
+  getDownloadURL,
+  uploadBytes,
+  ref as storageRef,
+} from "firebase/storage";
+import { database, storage } from "./firebase";
+
+const DB_MESSAGES_KEY = "messages";
+const DB_IMAGES_KEY = "images";
+
+export default function Composer({ uid, email }) {
+  const [inputValue, setInputValue] = useState("");
+  const [file, setFile] = useState(null);
+
+  const messagesRef = databaseRef(database, DB_MESSAGES_KEY);
+
+  const writeData = async () => {
+    let name = "";
+    let url = "";
+    if (file) {
+      const newStorageRef = storageRef(
+        storage,
+        DB_IMAGES_KEY + "/" + file.name
+      );
+      await uploadBytes(newStorageRef, file);
+      url = await getDownloadURL(newStorageRef);
+      name = file.name;
+    }
+    set(push(messagesRef), {
```

Comment:
Great implementation of likes to ensure that users can only like once, I also like that you have a like Count such that you dont need to process firebase data every load.